spec/activerecord-bitemporal/callbacks_spec.rb

```diff
@@ -0,0 +1,177 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+module BitemporalCallbacksLoggable
+  extend ActiveSupport::Concern
+
+  included do
+    attr_accessor :log
+    after_initialize { @log = [] }
+
+    [:before_bitemporal_create, :after_bitemporal_create,
+     :before_bitemporal_update, :after_bitemporal_update,
+     :before_bitemporal_destroy, :after_bitemporal_destroy
+    ].each do |callback_method|
+      send(callback_method, Proc.new { |model| model.log << callback_method })
+    end
+  end
+end
+
+class EmployeeWithBitemporalCallbaksLog < Employee
+  include BitemporalCallbacksLoggable
+  has_one :address, class_name: 'AddressWithBitemporalCallbacksLog', foreign_key: :employee_id
+end
+
+class EmployeeWithDependentDestroy < Employee
+  include BitemporalCallbacksLoggable
+  has_one :address, class_name: 'AddressWithBitemporalCallbacksLog', foreign_key: :employee_id, dependent: :destroy
+end
+
+class AddressWithBitemporalCallbacksLog < Address
+  include BitemporalCallbacksLoggable
+end
+
+RSpec.describe ActiveRecord::Bitemporal::Callbacks do
+  describe "bitemporal_create" do
+    let(:employee) { EmployeeWithBitemporalCallbaksLog.new(name: "Jane") }
+    subject { -> { employee.save! } }
+
+    context 'without option `bitemporal_callbacks`' do
+      it { is_expected.to change(employee, :log).from([]).to(%i[before_bitemporal_create after_bitemporal_create]) }
+    end
+
+    context 'with option `bitemporal_callbacks: true`' do
+      before { employee.bitemporal_option_merge!(bitemporal_callbacks: true) }
+      it { is_expected.to change(employee, :log).from([]).to(%i[before_bitemporal_create after_bitemporal_create]) }
+    end
+
+    context 'with option `bitemporal_callbacks: false`' do
+      before { employee.bitemporal_option_merge!(bitemporal_callbacks: false) }
+      it { is_expected.not_to change(employee, :log) }
+    end
+
+    context 'nested_attributes' do
+      let(:employee) {
+        Class.new(EmployeeWithBitemporalCallbaksLog) {
+          accepts_nested_attributes_for :address
+        }.new(name: "Jane").tap { |e| e.build_address(city: "Tokyo") }
+      }
+
+      context 'without option `bitemporal_callbacks`' do
+        it { is_expected.to change(employee, :log).from([]).to(%i[before_bitemporal_create after_bitemporal_create]) }
+        it { is_expected.to change { employee.address.log }.from([]).to(%i[before_bitemporal_create after_bitemporal_create]) }
+      end
+
+      context 'employee with option `bitemporal_callbacks: false`' do
+        before { employee.bitemporal_option_merge!(bitemporal_callbacks: false) }
+        it { is_expected.not_to change(employee, :log) }
+        it { is_expected.to change { employee.address.log }.from([]).to(%i[before_bitemporal_create after_bitemporal_create]) }
```

Comment:
We should consider whether it makes sense for the parent's bitemporal option not to propagate to the child. For example, the `valid_datetime` is propagated. This is a lack of consistency.

```ruby
employee.bitemporal_option_merge!(valid_datetime: '2023/02/01')
employee.save!

employee.valid_from         # => 2023/02/01
employee.address.valid_from # => 2023/02/01
```

Reply:
Thanks for the comment. 
Other bitemporal options (e.g. force_update) are not propagated to children now. Actually, valid_datetime option is not propagated too when children were loaded before set valid_datetime option.

```rb
# if already loaded employee.address
employee.bitemporal_option_merge!(valid_datetime: '2023/02/01')
employee.save!

employee.valid_from         # => 2023/02/01 00:00:00
employee.address.valid_from # => 2023/02/01 15:34:10 (Time.current)
```

I think it's better to open a new issue if we need to fix this problem because it has a large impact.